addons/website/tests/test_performance.py

```diff
@@ -53,35 +44,45 @@ def _get_url_hot_query(self, url, cache=True, table_count=False):
         self.url_open(url)
         self.url_open(url)
         self.url_open(url)
-        sql_count_before = self.cr.sql_log_count
-
-        with (self.cr._enable_table_tracking() if table_count else nullcontext()):
-            if table_count:
-                sql_from_log_before = copy.deepcopy(self.cr.sql_from_log)
-                sql_into_log_before = copy.deepcopy(self.cr.sql_into_log)
 
+        NestedProfiler = self.profile(collectors=['sql'], db=False)
+        with NestedProfiler:
+            self.registry.get_sequences(self.cr)
             self.url_open(url)
-            sql_count = self.cr.sql_log_count - sql_count_before - EXTRA_REQUEST
-            if table_count:
-                sql_from_tables = {'base_registry_signaling': 1}  # see EXTRA_REQUEST
-                sql_into_tables = {}
-                for table, stats in self.cr.sql_from_log.items():
-                    query_count = stats[0] - sql_from_log_before.get(table, [0])[0]
-                    if query_count:
-                        sql_from_tables[table] = query_count
-                for table, stats in self.cr.sql_into_log.items():
-                    query_count = stats[0] - sql_into_log_before.get(table, [0])[0]
-                    if query_count:
-                        sql_into_tables[table] = query_count
-                return sql_count, sql_from_tables, sql_into_tables
 
+        profiler = NestedProfiler.profiler
+        self.assertEqual(len(profiler.sub_profilers), 1, "we expect to have only one accessed url") # if not adapt the code below
+        route_profiler = profiler.sub_profilers[0]
+        route_entries = route_profiler.collectors[0].entries
+        entries = profiler.collectors[0].entries + route_entries
+        sql_queries = [entry['full_query'].strip() for entry in entries]
+        sql_count = len(sql_queries)
+        if not query_list:
             return sql_count
+        return sql_count, sql_queries
 
     def _check_url_hot_query(self, url, expected_query_count, select_tables_perf=None, insert_tables_perf=None):
-        query_count, select_tables, insert_tables = self._get_url_hot_query(url, table_count=True)
-        self.assertEqual(query_count, expected_query_count)
-        self.assertEqual(select_tables, select_tables_perf or {})
-        self.assertEqual(insert_tables, insert_tables_perf or {})
+        query_count, sql_queries = self._get_url_hot_query(url, query_list=True)
+
+        sql_from_tables = {}
+        sql_into_tables = {}
+
+        query_separator = '\n' + '-' * 100 + '\n'
+        queries = query_separator.join(sql_queries)
+
+        for query in sql_queries:
+            query_type, table = categorize_query(query)
+            if query_type == 'into':
+                log_target = sql_into_tables
+            elif query_type == 'from':
+                log_target = sql_from_tables
```

Comment:
What if `query_type` is `'other'`?  Your code will either crash with an undefined variable, or increment the former case.